plans/example-rust/src/main.rs

```diff
@@ -7,51 +8,43 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let (client, _run_parameters) = testground::client::Client::new().await?;
     client.wait_network_initialized().await?;
 
-    let local_addr = &if_addrs::get_if_addrs()
-        .unwrap()
+    let local_addr = &if_addrs::get_if_addrs()?
         .into_iter()
         .find(|iface| iface.name == "eth1")
-        .unwrap()
+        .ok_or("Could not find interface eth1")?
         .addr
         .ip();
 
     match local_addr {
         std::net::IpAddr::V4(addr) if addr.octets()[3] == 2 => {
             println!("Test instance, listening for incoming connections.");
 
-            let listener = TcpListener::bind((*addr, LISTENING_PORT))?;
+            let listener = TcpListener::bind((*addr, LISTENING_PORT)).await?;
 
             client.signal("listening".to_string()).await?;
 
-            for _stream in listener.incoming() {
-                println!("Established inbound TCP connection.");
-                break;
-            }
+            println!("Established inbound TCP connection.");
+            let _ = listener.accept().await?;
         }
         std::net::IpAddr::V4(addr) if addr.octets()[3] == 3 => {
             println!("Test instance, connecting to listening instance.");
 
-            client
-                .barrier("listening".to_string(), 1)
-                .await?;
+            client.barrier("listening".to_string(), 1).await?;
 
             let remote_addr: Ipv4Addr = {
                 let mut octets = addr.octets();
                 octets[3] = 2;
                 octets.into()
             };
-            let _stream = TcpStream::connect((remote_addr, LISTENING_PORT)).unwrap();
-            println!("Established outbound TCP connection.");
```

Comment:
Why drop the `println` on the dialer side?

Reply:
No reason, I switched the `println!`'s for `client.record_message` to understand what would be the difference on the logs and then missed this when reverting thanks for catching that Max.  

I noticed it's that `println!`'s use the `OTHER` scope while `client.record_message` uses the `MESSAGE` scope in the logs. Is there a place where the difference between using one or the other is explained? 
[`record_message`](https://docs.rs/testground/0.4.0/testground/client/struct.Client.html#method.record_message) doc seems to be missing a description also, I can then update it.

Reply:
Not really, 
`record_message` sends structured messages to the sync service, it should be capable to hold more info (like log severity, etc)
`println!` is "just" captured by testground and forwarded to the client.

Related: https://github.com/testground/testground/issues/1355